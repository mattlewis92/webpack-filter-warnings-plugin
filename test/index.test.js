import FilterWarningsPlugin from '../src/cjs';
import webpack from './utils/webpack';

test('filter warnings by regexp', async () => {
  const stats = await webpack({
    extend: {
      plugins: [
        new FilterWarningsPlugin({ exclude: /hide/ }),
      ],
    },
  });
  expect(stats.compilation.warnings).toMatchSnapshot();
});

test('filter warnings by function', async () => {
  const stats = await webpack({
    extend: {
      plugins: [
        new FilterWarningsPlugin({
          exclude: ({ message }) => message.indexOf('hide me') > -1,
        }),
      ],
    },
  });
  expect(stats.compilation.warnings).toMatchSnapshot();
});

test('allow array of regexp to filter by', async () => {
  const stats = await webpack({
    extend: {
      plugins: [
        new FilterWarningsPlugin({ exclude: [/hide/] }),
      ],
    },
  });
  expect(stats.compilation.warnings).toMatchSnapshot();
});

test('allow filter warnings that are strings', async () => {
  const stats = await webpack({
    extend: {
      plugins: [
        {
          apply(compiler) {
            compiler.hooks.done.tap('filter-warnings-plugin', (result) => {
              result.compilation.warnings.push('hide this string');
            });
          },
        },
        new FilterWarningsPlugin({ exclude: /hide/ }),
      ],
    },
  });
  expect(stats.compilation.warnings).toMatchSnapshot();
});
