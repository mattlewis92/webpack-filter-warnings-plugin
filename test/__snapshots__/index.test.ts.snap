// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main library file Function exclude should filter warnings based on given function 1`] = `
Array [
  [ModuleWarning: Module Warning (from ./test/utils/loader.ts):
show me],
]
`;

exports[`Main library file RegExp exclude should filter warnings by array of RegExps 1`] = `
Array [
  [ModuleWarning: Module Warning (from ./test/utils/loader.ts):
show me],
]
`;

exports[`Main library file RegExp exclude should filter warnings by regexp 1`] = `
Array [
  [ModuleWarning: Module Warning (from ./test/utils/loader.ts):
show me],
]
`;

exports[`Main library file String exclude should filter warnings based on string option 1`] = `
Array [
  [ModuleWarning: Module Warning (from ./test/utils/loader.ts):
show me],
]
`;

exports[`Main library file Type guards should throw when invalid arguments are passed 1`] = `[Error: Exclude can only be string, function, RegExp, or an Array of these.]`;
